### DISCUSSION TITLE
Difference between .scalars().all() and list(...scalars())

### DISCUSSION THREAD

**Discussion by abrahammurciano** [sqlalchemy/sqlalchemy#11569] [Usage Questions] [answered]
Are there any significant differences (performance, or otherwise) between the following two ways of getting a list of results?
```py
users: Sequence[User] = session.execute(select(User)).scalars().all()
```
```py
users: Sequence[User] = list(session.execute(select(User)).scalars())
```
The second seems more pythonic to me, and is possible because `ScalarResult` is iterable over its rows, so calling `list(...scalars())` just iterates over the `ScalarResult` object. However, since `.all()` also exists, I assume there must be some reason for it, so maybe it's more efficient to use `.all()`?

What are the differences between the two, and what, if anything, goes on behind the scenes that causes the difference?

**Comment by CaselIT** [comment, marked as answer]
Hi,

`all()` has the possibility of being more efficient since it can process all rows at a time instead of having to call fetchone many times while iterating. 
But overall the difference should not be too significant, compared with the query execution time
